contracts/ReputationMiningCycle.sol

```diff
@@ -147,38 +577,47 @@ contract ReputationMiningCycle {
       require(disputeRounds[round].length>1);
       // Move opponent on to next round
       disputeRounds[round+1].push(disputeRounds[round][opponentIdx]);
+      delete disputeRounds[round][opponentIdx];
```

Comment:
Additionally we can shrink the `array.length` http://solidity.readthedocs.io/en/v0.4.21/types.html?length#members
This might solve the note here https://github.com/JoinColony/colonyNetwork/pull/189/files#diff-21ee0c070df2d15abfdf2c1a0e8c8737R75

Reply:
That note, which now lives [here](https://github.com/JoinColony/colonyNetwork/pull/189/files#diff-21ee0c070df2d15abfdf2c1a0e8c8737R87) after rearranging the file in question, is warning against exactly what you're talking about here! That note may well be overcautious, because `delete` causing the array length to change would break so much other stuff, but I put it in because it is somewhat subtle behaviour that we are relying on for correct functioning of this contract, that isn't apparent at first glance. 

After the dispute process has been completed, the last two `disputeRounds` could be, if Round5 started as `[b, a]` and length was updated

```
Round5: [b]
Round6: [a]
```

The modifier `finalDisputeRoundCompleted` will now pass with both `5` and `6`, because the length of Round5 was update.

In any case, I actually don't think we can safely reduce `array.length` here - we will only rarely be deleting the last element in an array, which is the scenario where you can change `array.length` and not lose access to anything, right?